Controller/Adminhtml/Cart/Data.php

```diff
@@ -101,12 +133,33 @@ public function execute()
     {
         $startTime = $this->metricsClient->getCurrentTime();
         try {
-            // call the Bolt API
-            $boltpayOrder = $this->cartHelper->getBoltpayOrder(true, '');
+            $storeId = $this->_getSession()->getStoreId();
+            if (!$storeId) {
+                throw new LocalizedException(__('Order creation not initialized'));
+            }
+            $this->_initSession();
+
+            $quote = $this->_getOrderCreateModel()->getQuote();
+            $this->storeManager->setCurrentStore($storeId);
+
+            $backOfficeKey = $this->configHelper->getPublishableKeyBackOffice();
+            $paymentOnlyKey = $this->configHelper->getPublishableKeyPayment();
+            $isPreAuth = $this->configHelper->getIsPreAuth();
+
+            $customerEmail = $quote->getCustomerEmail();
+            if (!$quote->getCustomerId() && $this->cartHelper->getCustomerByEmail($customerEmail)) {
```

Comment:
Do I understand correctly, If we try to create backoffice order for guest user but with email that belongs to any m2 user we want to generate exception? Why we want it? Which error eventually merchant user sees and in which moment?

Reply:
@vitaliyreznikov
How we did it so far is always creating a back-office order as a guest user, and never taking care of really creating a customer account and putting it into appropriate customer group. It was kind of hard-coded in the module. That's not how Magento does it and it's very strange that we missed it and no merchant pointed that out until now.  

Basically, in the back-office create order flow either an existing customer is selected or a new one needs to be created in the process (_Create New Customer_ button). No guests in Magento back-office orders. And all our back-office order customers in the create new customer flow were guests so far. 

If you chose an alternative payment method (e.g. Check / Money order) with create new customer option, enter existing user email, fill the form data (shipping...) and try to _Submit order_, you will get this error:

**_A customer with the same email address already exists in an associated website._**

Reply:
I am starting to understand. Do you mind add to PR description: which issue we try to solve here; exact steps to reproduce.
It can be helpful when QA team will test this during the release cycle.

Reply:
done